artichoke-backend/src/class_registry.rs

```diff
@@ -76,7 +47,7 @@ impl ClassRegistry for Artichoke {
             self.with_ffi_boundary(|mrb| {
                 if let Some(mut rclass) = rclass.resolve(mrb) {
                     let value_class = sys::mrb_sys_class_value(rclass.as_mut());
-                    Some(Value::from(value_class))
+                    Some(Self::Value::from(value_class))
```

Comment:
I like using `Self::Value` in the function signatures to match the trait definition, but in the function body, I'd prefer if we use the concrete type instead of indirecting through the trait.

```suggestion
                    Some(Value::from(value_class))
```